pkg/unvanquished_src.dpkdir/bots/default_aliens.bt

```diff
@@ -1,5 +1,48 @@
 selector
 {
+	sequence
+	{
+		condition levelTime >= 60000
+		condition numUsersInTeam == 0 || numOurSpawns < 2
+		condition usableBuildPoints >= chosenBuildableCost
+		condition !teamateIsClass( PCL_ALIEN_BUILDER0 ) && !teamateIsClass( PCL_ALIEN_BUILDER0_UPG )
+		selector
+		{
+			sequence
+			{
+				condition class != PCL_ALIEN_BUILDER0 && class != PCL_ALIEN_BUILDER0_UPG
+				selector
+				{
+					condition timeSinceMyTimerUpdate >= 20000
+					{
+						action suicide
```

Comment:
If suicide is going to be used to reach the builder state, you really need some coordination so I think it would be better to use the change BT thing like you said you had on the server. Otherwise if there is no OM or no armory, the bots are all going to suicide simultaneously.

Alternatively the suicide bit could be removed. 20 seconds is a kind of long time to go without anyone on the team being killed anyway

Reply:
What I did on my server turned out to be confusing to users and admins. I have come to believe that doing these things in default.bt is preferrable to changing BTs by backgroung magic.

I think a better way would be to make a general purpose mechanism for bots to share such information. Some suggested we call this a "blackboard".

Reply:
I gave each bot a "transient" field on the "blackboard", that is: an integer in its mind that is reset to zero at each frame. I also added a function to make bots the non-zero fields in their team. This is enough to ensure that only one living bot tries to become the builder.

Reply:
Notice that, if no BT change is ever done automatically, we can allow any BT changes using the tactic command. No confusion will arise.